.github/workflows/publish.yml

```diff
@@ -0,0 +1,81 @@
+name: Publish and deploy
+on: create
+
+env:
+  SERVICE_NAME: ${{ github.event.repository.name }}
+  GOOGLE_PROJECT: broad-dsp-gcr-public
+
+jobs:
+  publish-job:
+    # only publish on new tag
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-refs/heads/dev-${{ github.sha }}
```

Comment:
I'm wondering about what this job is doing, since we don't use it in the catalog workflow code. Does it speed up the build? I remember looking at gradle build caching when I was setting up catalog and ended up looking at https://github.com/burrunan/gradle-cache-action but haven't done anything with it yet.

Reply:
I wonder how much improvement you'll get. I guess that would depend on how much time it takes to build in the first place.